src/modules/storage/LedgerStorage.ts

```diff
@@ -907,20 +907,54 @@ export class LedgerStorage extends Storages {
             });
         }
 
+        function updateValidatorSlashing(storage: LedgerStorage, height: Height, utxo_key: Hash, enrolled_at: Height) {
+            return new Promise<void>((resolve, reject) => {
+                storage
+                    .query(
+                        `UPDATE validators
+                            SET slashed = ? ,
+                            slashed_height = ?
+                            WHERE utxo_key = ? AND enrolled_at = ?`,
+                        [
+                            ValidatorStatus.SLASHED,
+                            height.value.toString(),
+                            utxo_key.toBinary(Endian.Little),
+                            enrolled_at,
+                        ],
+                        conn
+                    )
+                    .then(() => {
+                        resolve();
+                    })
+                    .catch((err) => {
+                        reject(err);
+                    });
+            });
+        }
+
         return new Promise<void>((resolve, reject) => {
             (async () => {
                 const validators: any[] = await this.getValidatorsAPI(block.header.height, null, conn);
                 for (let idx = 0; idx < block.header.preimages.length; idx++) {
+                    if (idx >= validators.length) continue;
+                    if (validators[idx] === undefined) continue;
                     try {
-                        if (undefined !== validators[idx] && block.header.preimages[idx] !== new Hash(Buffer.alloc(Hash.Width))) {
-                            await save_preimage(
+                        if (Buffer.compare(block.header.preimages[idx].data, Buffer.alloc(Hash.Width)) === 0) {
```

Comment:
```suggestion
                        if (Hash.isNull(block.header.preimages[idx])) {
```
Please raise the version of SDK to 0.2.6, and I hope you can modify the code like this.

Reply:
I applied the suggestion.

Reply:
The suggestion above is wrong.
I think it'd be good to do it like this.
```suggestion
                        if (block.header.preimages[idx].isNull()) {
```

Reply:
I applied the suggestion.